gpkit/constraints/prog_factories.py

```diff
@@ -0,0 +1,115 @@
+            else:
+                sweep_grids = np.meshgrid(*list(sweep.values()))
+
+            N_passes = sweep_grids[0].size
+            sweep_vects = {var: grid.reshape(N_passes)
+                           for (var, grid) in zip(sweep, sweep_grids)}
+
+            if verbosity > 1:
+                print("Solving over %i passes." % N_passes)
+                tic = time()
+
+            def solve_pass(i):
+                this_pass = {var: sweep_vect[i]
+                             for (var, sweep_vect) in sweep_vects.items()}
+                linked = {var: fn(*[this_pass[VarKey(v)]
+                                    for v in var.descr["args"]])
+                          for var, fn in linkedsweep.items()}
+                this_pass.update(linked)
+                constants.update(this_pass)
+                program, solvefn = genfunction(self, verbosity-2, constants)
+                try:
+                    result = solvefn(solver, verbosity-2, *args, **kwargs)
+                    # add localmodel here
+                    return program, result
+                except (RuntimeWarning, ValueError):
+                    return program, None
+
+            if POOL:
+                mapfn = POOL.map_sync
+            else:
+                mapfn = map
+
+            self.program = []
+            for program, result in mapfn(solve_pass, range(N_passes)):
+                self.program.append(program)  # NOTE: SIDE EFFECTS
+                if result:  # solve succeeded
+                    solution.append(result)
+                elif not kwargs.get("skipsweepfailures"):
+                    raise RuntimeWarning("solve failed during sweep; program"
+                                         " has been saved to m.program[-1]."
+                                         " To ignore such failures, solve with"
+                                         " skipsweepfailures=True.")
+
+            solution["sweepvariables"] = KeyDict()
+            ksweep, klinkedsweep = KeyDict(sweep), KeyDict(linkedsweep)
+            for var, val in solution["constants"].items():
+                if var in ksweep or var in klinkedsweep:
+                    solution["sweepvariables"][var] = val
+                    del solution["constants"][var]
+                else:
+                    solution["constants"][var] = [val[0]]
+            if not solution["constants"]:
+                del solution["constants"]
+
+            if verbosity > 1:
+                soltime = time() - tic
+                print("Sweeping took %.3g seconds." % (soltime,))
+        solution.program = self.program
+        solution.to_united_array(unitless_keys=["sensitivities"], united=True)
+        # solution["localmodel"] = NomialArray(solution["localmodel"])
+        self.solution = solution  # NOTE: SIDE EFFECTS
+        if verbosity > 0:
+            print(solution.table())
+        return solution
+    return solvefn
```

Comment:
At first glance, it seems to me more related to `goemetric_program.py`, which is a level up.